CDP4WebServices.API/Services/Operations/SideEffects/Implementation/ParticipantSideEffect.cs

```diff
@@ -0,0 +1,75 @@
+
+namespace CDP4WebServices.API.Services.Operations.SideEffects
+{
+    using System;
+
+    using Authorization;
+    using CDP4Common;
+    using CDP4Common.DTO;
+    using Npgsql;
+
+    /// <summary>
+    /// The purpose of the <see cref="ParticipantSideEffect"/> class is to execute additional logic before and after a specific operation is performed.
+    /// </summary>
+    public class ParticipantSideEffect : OperationSideEffect<Participant>
+    {
+        /// <summary>
+        /// The selected domain key.
+        /// </summary>
+        private const string SelectedDomainKey = "SelectedDomain";
+
+        /// <summary>
+        /// Allows derived classes to override and execute additional logic before an update operation.
+        /// </summary>
+        /// <param name="thing">
+        /// The <see cref="Thing"/> instance that will be inspected.
+        /// </param>
+        /// <param name="container">
+        /// The container instance of the <see cref="Thing"/> that is inspected.
+        /// </param>
+        /// <param name="transaction">
+        /// The current transaction to the database.
+        /// </param>
+        /// <param name="partition">
+        /// The database partition (schema) where the requested resource will be stored.
+        /// </param>
+        /// <param name="securityContext">
+        /// The security Context used for permission checking.
+        /// </param>
+        /// <param name="rawUpdateInfo">
+        /// The update info that was serialized directly from the user request. 
+        /// The raw <see cref="ClasslessDTO"/> instance only contains values for properties that are to be updated.
+        /// It is important to note that this variable is not to be edited likely: it can/will change the operation processor outcome.
+        /// </param>
+        public override void BeforeUpdate(Participant thing, Thing container, NpgsqlTransaction transaction, string partition, ISecurityContext securityContext, ClasslessDTO rawUpdateInfo)
+        {
+            this.ValidateSelectedDomain(thing, rawUpdateInfo);
+        }
+
+        /// <summary>
+        /// Checks whether a valid selected domain is supplied <see cref="Participant"/>.
+        /// </summary>
+        /// <param name="thing">
+        /// The <see cref="Thing"/> instance that will be inspected.
+        /// </param>
+        /// <param name="rawUpdateInfo">
+        /// The update info that was serialized directly from the user request. 
+        /// The raw <see cref="ClasslessDTO"/> instance only contains values for properties that are to be updated.
+        /// It is important to note that this variable is not to be edited likely: it can/will change the operation processor outcome.
+        /// </param>
+        private void ValidateSelectedDomain(Participant thing, ClasslessDTO rawUpdateInfo)
+        {
+            Guid.TryParse(rawUpdateInfo[SelectedDomainKey].ToString(), out var selectedDomainUpdate);
```

Comment:
Whats the advantage of using a const for this instead of nameof(Participant.SelectedDomain)?

Reply:
I keep the key for a better description.

Reply:
I think it's overhead, but that's my opinion.